code/obj/artifacts/artifact_items/agitator.dm

```diff
@@ -0,0 +1,43 @@
+/obj/item/artifact/agitator
+	name = "artifact agitator"
+	associated_datum = /datum/artifact/agitator
+
+	afterattack(atom/target, mob/user, reach, params)
+		var/datum/artifact/agitator/src_art = src.artifact
+		if (!src_art.activated)
+			return ..()
+		var/obj/O = target
+		if (!istype(O) || !O.artifact)
+			return ..()
+		if (!O.artifact.activated)
+			return ..()
+		. = ..()
+		if (ON_COOLDOWN(src, "artifact_agitation", rand(30, 180) SECONDS))
```

Comment:
Most artifacts like this randomize their cooldown once and save it to the artifact datum, this probably should too?